musit-service/src/main/scala/no/uio/musit/security/crypto/MusitCrypto.scala

```diff
@@ -0,0 +1,134 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package no.uio.musit.security.crypto
+
+import java.security.MessageDigest
+import javax.crypto.Cipher
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+
+import com.google.inject.{Inject, Singleton}
+import org.apache.commons.codec.binary.Base64
+import play.api.Logger
+import play.api.libs.Codecs
+import play.api.libs.crypto.CryptoException
+
+/**
+ * This class contains a copy of the deprecated implementation of
+ * {{{play.api.libs.crypto.AESCTRCrypter}}}. The code is duplicated because we
+ * need symmetric encryption for a specific use-case in the admin client.
+ *
+ * When time allows, the admin client should be extracted as a separate app,
+ * and modified to use the REST endpoints to load data.
+ */
+@Singleton
+class MusitCrypto @Inject() (ccp: CryptoConfigParser) {
+
+  val logger = Logger(classOf[MusitCrypto])
+
+  lazy val config = ccp.get
+
+  private val cipher = Cipher.getInstance(config.aesTransformation)
+
+  def encryptAES(value: String): String = {
+    encryptAES(value, config.secret)
+  }
+
+  def encryptAES(value: String, privateKey: String): String = {
+    logger.trace(s"Encrypting value: $value")
+    val skeySpec = secretKeyWithSha256(privateKey, "AES")
+    cipher.init(Cipher.ENCRYPT_MODE, skeySpec)
+    val encryptedValue = cipher.doFinal(value.getBytes("utf-8"))
+    // return a formatted, versioned encrypted string
+    // '2-*' represents an encrypted payload with an IV
+    // '1-*' represents an encrypted payload without an IV
+    logger.trace(s"Using IV ${Option(cipher.getIV).map(_.mkString)}")
+    val enc = Option(cipher.getIV) match {
+      case Some(iv) => s"2-${Base64.encodeBase64String(iv ++ encryptedValue)}"
+      case None => s"1-${Base64.encodeBase64String(encryptedValue)}"
+    }
+    logger.trace(s"Encrypted value is $enc")
+    enc
+  }
+
+  /**
+   * Generates the SecretKeySpec, given the private key and the algorithm.
+   */
+  private def secretKeyWithSha256(privateKey: String, algorithm: String) = {
+    val messageDigest = MessageDigest.getInstance("SHA-256")
+    messageDigest.update(privateKey.getBytes("utf-8"))
+    // max allowed length in bits / (8 bits to a byte)
+    val maxAllowedKeyLength = Cipher.getMaxAllowedKeyLength(algorithm) / 8
```

Comment:
Det stemmer, og det er også en av grunnene til at krypto biten i Play er deprecated. Men for våre formål er det greit nok.